pkg/query-service/app/traces/v4/query_builder.go

```diff
@@ -0,0 +1,422 @@
+package v4
+
+import (
+	"fmt"
+	"strings"
+
+	"go.signoz.io/signoz/pkg/query-service/app/resource"
+	tracesV3 "go.signoz.io/signoz/pkg/query-service/app/traces/v3"
+	"go.signoz.io/signoz/pkg/query-service/constants"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/utils"
+)
+
+const NANOSECOND = 1000000000
+
+var tracesOperatorMappingV3 = map[v3.FilterOperator]string{
+	v3.FilterOperatorIn:              "IN",
+	v3.FilterOperatorNotIn:           "NOT IN",
+	v3.FilterOperatorEqual:           "=",
+	v3.FilterOperatorNotEqual:        "!=",
+	v3.FilterOperatorLessThan:        "<",
+	v3.FilterOperatorLessThanOrEq:    "<=",
+	v3.FilterOperatorGreaterThan:     ">",
+	v3.FilterOperatorGreaterThanOrEq: ">=",
+	v3.FilterOperatorLike:            "ILIKE",
+	v3.FilterOperatorNotLike:         "NOT ILIKE",
+	v3.FilterOperatorRegex:           "match(%s, %s)",
+	v3.FilterOperatorNotRegex:        "NOT match(%s, %s)",
+	v3.FilterOperatorContains:        "ILIKE",
+	v3.FilterOperatorNotContains:     "NOT ILIKE",
+	v3.FilterOperatorExists:          "mapContains(%s, '%s')",
+	v3.FilterOperatorNotExists:       "NOT has(%s%s, '%s')",
+}
+
+func getClickHouseTracesColumnType(columnType v3.AttributeKeyType) string {
+	if columnType == v3.AttributeKeyTypeResource {
+		return "resources"
+	}
+	return "attributes"
+}
+
+func getClickHouseTracesColumnDataType(columnDataType v3.AttributeKeyDataType) string {
+	if columnDataType == v3.AttributeKeyDataTypeFloat64 || columnDataType == v3.AttributeKeyDataTypeInt64 {
+		return "number"
+	}
+	if columnDataType == v3.AttributeKeyDataTypeBool {
+		return "bool"
+	}
+	return "string"
+}
+
+func getColumnName(key v3.AttributeKey) string {
+	if !key.IsColumn {
+		keyType := getClickHouseTracesColumnType(key.Type)
+		keyDType := getClickHouseTracesColumnDataType(key.DataType)
+		return fmt.Sprintf("%s_%s['%s']", keyType, keyDType, key.Key)
+	}
+
+	// check if it is a static field
+	if key.Type == v3.AttributeKeyTypeUnspecified {
+		// name is the column name
+		return key.Key
+	}
```

Comment:
`AttributeKeyTypeUnspecified` doesn't mean static?

Reply:
It's kind of dead code here, in logs we do `and` with the static fields. Let me remove this out for now.